# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmStubs.cpp
# Commit: 7502fc7d616a
# Full Hash: 7502fc7d616a153f13516aafaa33acdb621222d2
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-04-09 09:55:00
# Regressor Bug: 1620197
# File Overlap Count: 1
# Description:
#   Bug 1624886 - Fix argBytes calculation in GenerateImportInterpExit. r=lth
#   
#   Bug 1620197 accidentally moved the parentheses for 'max' outside of the
#   multiplication by sizeof(Value).
#   
# ==============================================================================

diff -r 70b0b35496a5 -r 7502fc7d616a js/src/wasm/WasmStubs.cpp
--- a/js/src/wasm/WasmStubs.cpp	Wed Apr 08 16:17:50 2020 +0000
+++ b/js/src/wasm/WasmStubs.cpp	Wed Apr 08 13:25:18 2020 +0000
@@ -1889,7 +1889,7 @@
       AlignBytes(StackArgBytes(invokeArgTypes), sizeof(double));
   // The abiArgCount includes a stack result pointer argument if needed.
   unsigned abiArgCount = ArgTypeVector(fi.funcType()).length();
-  unsigned argBytes = std::max<size_t>(1, abiArgCount * sizeof(Value));
+  unsigned argBytes = std::max<size_t>(1, abiArgCount) * sizeof(Value);
   unsigned framePushed =
       StackDecrementForCall(ABIStackAlignment,
                             sizeof(Frame),  // pushed by prologue
